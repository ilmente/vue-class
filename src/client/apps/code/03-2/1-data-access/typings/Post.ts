export interface Post {
    id: string;
    name: string;
    username: string;
    content: string;
    imageSrc: string;
    dislikes: number;
}
