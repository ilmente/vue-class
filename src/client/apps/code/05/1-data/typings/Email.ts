export enum EmailStatus {
    UNREAD = 'unread',
    READ = 'read',
    SPAM = 'spam',
}

export interface EmailInfo {
    id: string;
    sender: string;
    subject: string;
    timestamp: string;
    status: EmailStatus;
}

export interface Email extends EmailInfo {
    content: string;
}
