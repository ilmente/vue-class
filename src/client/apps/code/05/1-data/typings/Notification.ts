export enum NotificationType {
    Default = '',
    Info = 'info',
    Success = 'success',
    Warning = 'warning',
    Danger = 'danger',
}

export interface Notification {
    content: string;
    type: NotificationType;
}
