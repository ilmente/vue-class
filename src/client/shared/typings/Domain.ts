export enum Domain {
    docs = 'docs',
    code = 'code',
    exercise = 'exercise',
}
