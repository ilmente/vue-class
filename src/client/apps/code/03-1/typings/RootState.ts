import { InitialState } from '@typings/InitialState';
import { Post } from './Post';

export interface RootState extends InitialState {
    posts: Post[];
}
