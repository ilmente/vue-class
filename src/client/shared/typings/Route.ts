import { Domain } from './Domain';

export type RouteMap = {
    [key in Domain]: { template: string, resolve: (entry?: string) => string };
}
