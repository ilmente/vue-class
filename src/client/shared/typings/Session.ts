import { Domain } from './Domain';

export interface SessionItem {
    url: string;
    name?: string;
}

export interface Session {
    name: string;
    [Domain.docs]: SessionItem;
    [Domain.code]: SessionItem;
    [Domain.exercise]: SessionItem;
}
