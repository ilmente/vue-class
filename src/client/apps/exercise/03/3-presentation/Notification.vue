<template>
    <transition name="fade">
        <div v-show="isVisible" class="notification" :class="data.type && `is-${data.type}`">
            <button class="delete" @click="onClick"></button>
            <slot>{{data.content}}</slot>
        </div>
    </transition>
</template>

<style lang="scss" scoped>
    .fade-enter-active, 
    .fade-leave-active {
        transition: opacity 300ms;
    }
    .fade-enter, 
    .fade-leave-to {
        opacity: 0;
    }
</style>

<script lang="ts">
    import { Vue, Component, Prop } from 'vue-property-decorator';
    import { Notification } from '../1-data/typings/Notification';

    const DEFAULT_TIMEOUT = 5000;

    @Component({
        name: 'Notification',
    })
    export default class extends Vue {
        @Prop({
            type: Object,
            required: true,
        }) data!: Notification;

        timeoutId: ReturnType<typeof setTimeout> | null = null;
        isVisible: boolean = false;

        mounted(): void {
            this.isVisible = true;
            this.hide();
        }

        onClick(): void {
            if (this.timeoutId) {
                clearTimeout(this.timeoutId);
            }
            
            this.isVisible = false;
        }

        hide(): void {
            this.timeoutId = setTimeout(() => {
                this.isVisible = false;
            }, DEFAULT_TIMEOUT);
        }
    }
</script>
