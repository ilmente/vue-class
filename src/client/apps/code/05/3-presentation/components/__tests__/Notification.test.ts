import Notification from '../Notification.vue';
import { mount } from '@vue/test-utils';
import { Notification as INotification, NotificationType } from '../../../1-data/typings/Notification';

const notification: INotification = {
    content: 'This is a notification',
    type: NotificationType.Success,
}

const propsData = {
    data: notification,
}

describe('Notification.vue', () => {
    const wrapper = mount(Notification, {
        propsData,
    });

    test('snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    test('contains the right content', () => {
        const content = wrapper.find('.notification__content');
        expect(content.text()).toBe(notification.content);
    });

    test('display the right type', () => {
        const container = wrapper.find('.notification');
        expect(container.classes()).toContain(`is-${notification.type}`);
    });
});
