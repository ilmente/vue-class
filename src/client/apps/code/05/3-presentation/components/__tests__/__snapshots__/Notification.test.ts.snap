// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification.vue snapshot 1`] = `
<transition-stub
  name="fade"
>
  <div
    class="notification is-success"
    data-qa-notification=""
    style=""
  >
    <button
      class="delete"
    />
     
    <div
      class="notification__content"
    >
      This is a notification
    </div>
  </div>
</transition-stub>
`;
